esmvaltool/diag_scripts/weathertyping/weathertyping.py

```diff
@@ -0,0 +1,395 @@
+    data_info
+        Dictionary holding dataset information.
+    """
+    for ensemble_var in dataset_vars:
+        if ensemble_var.get("preprocessor") == "weathertype_preproc":
+            data_info["ensemble"] = ensemble_var.get("ensemble", "")
+            data_info["driver"] = ensemble_var.get("driver", "")
+
+            if data_info["driver"] != "":
+                data_info["driver"] = "_" + {data_info["driver"]}
+
+            wt_preproc = iris.load_cube(dataset_vars[0].get("filename"))
+
+            output_file_path, preproc_path = get_model_output_filepath(
+                data_info["dataset"],
+                dataset_vars,
+            )
+
+            data_info["output_file_path"] = output_file_path
+            data_info["preproc_path"] = preproc_path
+
+            # calculate weathertypes
+            calc_lwt_model(cfg, wt_preproc, data_info)
+
+            # load wt files
+            wt_cubes = load_wt_files(
+                f"{cfg.get('work_dir')}/{output_file_path}"
+                f"/{data_info['dataset']}"
+                f"{data_info['driver']}"
+                f"_{data_info['ensemble']}_"
+                f"{data_info['timerange']}.nc",
+                mode="lwt",
+            )
+
+            var_dict = get_looping_dict(
+                dataset_vars,
+            )  # dataset_vars is list of variables for dataset_name
+
+            if cfg.get("plotting", False):
+                # plot means
+                for var_name, var_data in var_dict.items():
+                    data_info["var"] = var_name
+                    data_info["preproc_path"] = var_data[1]
+
+                    plot_means(
+                        cfg,
+                        var_data[0],
+                        wt_cubes,
+                        data_info,
+                        "lwt",
+                    )
+                plot_seasonal_occurrence(cfg, wt_cubes, data_info)
+
+
+def run_lwt(cfg: dict):
+    """Run calculation of weathertypes. Write to file, and plot the means of psl, tas, and pr for each weathertype. Plot seasonal occurrence of the weathertypes.
+
+    Parameters
+    ----------
+    cfg
+        Nested dictionary of metadata.
+    """
+    preproc_variables_dict = group_metadata(
+        cfg.get("input_data").values(),
+        "dataset",
+    )
+    for dataset_name, dataset_vars in preproc_variables_dict.items():
+        data_info = {
+            "timerange": dataset_vars[0].get("timerange").replace("/", "-"),
+            "dataset": dataset_name,
+        }
+
+        if dataset_name == "ERA5":
+            process_era5_lwt(
+                preproc_variables_dict,
+                cfg,
+                dataset_vars,
+                data_info,
+            )
+        else:
+            if data_info["dataset"] == "E-OBS":
+                continue
+            process_models_lwt(cfg, dataset_vars, data_info)
+
+
+def run_my_diagnostic(cfg: dict):
```

Comment:
it'd be nice if you could call it something a bit more relevant eg `run_lwt_diagnostic`